ic-canister-client/src/pocket_ic.rs

```diff
@@ -50,10 +83,17 @@ impl PocketIcClient {
     {
         let args = candid::encode_args(args)?;
 
-        let reply = self
-            .client()
-            .update_call(self.canister, self.caller, method, args)
-            .await?;
+        let reply = if self.live {
```

Comment:
The `live` is not needed. From the code, it seems to me that if the client is in live mode then `client.url()` returns `Some(url)`. We can ask in the Dfinity forum whether this is the correct way to detect if a client is live.

Reply:
Good catch, that is Some only in live mode